homeassistant/components/knx/cover.py

```diff
@@ -73,99 +90,95 @@ def supported_features(self):
             return None
         return self._device.is_closed()
 
     @property
-    def is_opening(self):
+    def is_opening(self) -> bool:
         """Return if the cover is opening or not."""
         return self._device.is_opening()
 
     @property
-    def is_closing(self):
+    def is_closing(self) -> bool:
         """Return if the cover is closing or not."""
         return self._device.is_closing()
 
-    async def async_close_cover(self, **kwargs):
+    async def async_close_cover(self, **kwargs: Any) -> None:
         """Close the cover."""
         await self._device.set_down()
 
-    async def async_open_cover(self, **kwargs):
+    async def async_open_cover(self, **kwargs: Any) -> None:
         """Open the cover."""
         await self._device.set_up()
 
-    async def async_set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Move the cover to a specific position."""
         knx_position = 100 - kwargs[ATTR_POSITION]
         await self._device.set_position(knx_position)
 
-    async def async_stop_cover(self, **kwargs):
+    async def async_stop_cover(self, **kwargs: Any) -> None:
         """Stop the cover."""
         await self._device.stop()
         self.stop_auto_updater()
 
     @property
-    def current_cover_tilt_position(self):
+    def current_cover_tilt_position(self) -> int | None:
         """Return current tilt position of cover."""
         if not self._device.supports_angle:
             return None
-        try:
-            return 100 - self._device.current_angle()
-        except TypeError:
-            return None
+        ang = self._device.current_angle()
+        return 100 - ang if ang is not None else None
 
-    async def async_set_cover_tilt_position(self, **kwargs):
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
         """Move the cover tilt to a specific position."""
         knx_tilt_position = 100 - kwargs[ATTR_TILT_POSITION]
         await self._device.set_angle(knx_tilt_position)
 
-    async def async_open_cover_tilt(self, **kwargs):
+    async def async_open_cover_tilt(self, **kwargs: Any) -> None:
         """Open the cover tilt."""
         await self._device.set_short_up()
 
-    async def async_close_cover_tilt(self, **kwargs):
+    async def async_close_cover_tilt(self, **kwargs: Any) -> None:
         """Close the cover tilt."""
         await self._device.set_short_down()
 
-    async def async_stop_cover_tilt(self, **kwargs):
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
         """Stop the cover tilt."""
         await self._device.stop()
         self.stop_auto_updater()
 
-    def start_auto_updater(self):
+    def start_auto_updater(self) -> None:
         """Start the autoupdater to update Home Assistant while cover is moving."""
-        if self._unsubscribe_auto_updater is None:
+        if self._unsubscribe_auto_updater is None and self.hass is not None:
```

Comment:
This should not be needed.

```suggestion
        if self._unsubscribe_auto_updater is None:
```